specs/validator-system-v2/asset-manager.md

```diff
@@ -0,0 +1,223 @@
+even if rewards are generated from output finalization, they will not be retroactively applied.
+
+However, if slashing occurs due to a challenge, assets in the pending state will also be included in the slashing scope.
+This measure is to prevent abuse as mentioned in the second case. If pending assets were not subject to slashing,
+users could undelegate their assets immediately to avoid slashing regardless of the challenge outcome.
+
+This would reduce the rewards for the winner of the challenge, decrease the incentive to create challenges, and
+negatively impact the network's security. Therefore, assets in the pending state must be included within the slashing
+scope.
+
+To achieve this, assets in the pending state must be managed separately from assets delegated to the vault.
+Information related to this is stored in the `vault` within the following `Pending` struct:
+
+```solidity
+struct Pending {
+    uint128 totalPendingAssets;
+    uint128 totalPendingBoostedRewards;
+    uint128 totalPendingKroShares;
+    uint128 totalPendingKghShares;
+    uint128 totalPendingValidatorRewards;
+    uint256[] claimRequestTimes;
+    mapping(uint256 => uint128) pendingValidatorRewards;
+}
+```
+
+When a user initiates undelegation, the requested amount of assets and shares are stored separately.
+After the one-week delay ends, the total pending assets, the user's requested pending shares, and the total pending
+shares are used to calculate the final amount of assets to be received. The pending shares and undelegate request time
+information for each delegator, along with the shares of each delegator, are saved in the following structs:
+
+```solidity
+struct KroDelegator {
+    uint128 shares;
+    uint256[] undelegateRequestTimes;
+    mapping(uint256 => uint128) pendingKroShares;
+}
+
+struct KghDelegator {
+    mapping(uint256 => KghDelegatorShares) shares;
+    uint256[] undelegateRequestTimes;
+    mapping(uint256 => uint256[]) pendingKghIds;
+    mapping(uint256 => KghDelegatorShares) pendingShares;
+}
+```
+
+## Reward Distribution
+
+Every time a validator successfully submits an output root, the `_distributeReward` function within the
+`ValidatorManager` contract is called to distribute rewards to the respective validator's vault.
+The rewards are calculated as follows:
+
+| Rewards For KRO Delegators   | Rewards For KGH NFT Delegators   |
+|------------------------------|----------------------------------|
+| BASE_REWARD $\times (1-c_i)$ | BOOSTED_REWARD $\times (1-c_i)$  |
+
+Here, $c_i$ refers to the commission rate set during the validator registration process. The validator who has
+successfully submitted an output root receives extra reward corresponding to the commission rate, which is
+stored in the `validatorKro` within the `Asset` struct. Similar to undelegation, claiming a validator's
+reward is subject to a one-week delay, during which the claimed amount can be subject to slashing.
```

Comment:
I think the calculation of reward is part of ValidatorManager, so it's more appropriate to move this part to ValidatorManager. And how about just adding here how base reward and boosted reward, validator reward are stored?